lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -541,6 +543,29 @@ def confirm_namespace_exists
       @logger.info("Namespace #{@namespace} found")
     end
 
+    # make sure to never prune the ejson-keys secret
+    def check_applied_ejson_keys_secret
+      raw_secret, st, err = nil
+      with_retries(2) do
+        raw_secret, err, st = kubectl.run("get", "-n", @namespace, "secret",
+          "ejson-keys", "-oyaml", output_is_sensitive: true)
+        st.success? || err.include?(KubernetesDeploy::Kubectl::NOT_FOUND_ERROR_TEXT)
+      end
+      if st.success?
+        secret = YAML.safe_load(raw_secret, [Date, Time])
+        return unless secret.dig("metadata", "annotations", LAST_APPLIED_ANNOTATION)
+        @logger.warn("Secret ejson-keys has annotation #{LAST_APPLIED_ANNOTATION}.")
+        @logger.warn("kubernetes-deploy will remove this annotation to ensure this secret does not get pruned.")
+        with_retries(2) do
+          _, err, st = kubectl.run("annotate", "secret", "ejson-keys", "#{LAST_APPLIED_ANNOTATION}-")
```

Comment:
- same comment about using the constant for the name
- the output of this command likely isn't sensitive in practice, but since we know we are dealing with a secret, maybe we should use that option anyway out of an abundance of precaution